2022Swerve/src/main/java/frc/robot/commands/SwerveTeleopCommand.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.commands;
+
+import java.util.function.Supplier;
+
+import edu.wpi.first.math.filter.SlewRateLimiter;
+import edu.wpi.first.math.kinematics.ChassisSpeeds;
+import edu.wpi.first.math.kinematics.SwerveModuleState;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import frc.robot.Constants.DriveConstants;
+import frc.robot.Constants.OIConstants;
+import frc.robot.subsystems.SwerveSubsystem;
+
+public class SwerveTeleopCommand extends CommandBase {
+
+    private final SwerveSubsystem swerveSubsystem;
+    private final Supplier<Double> xSpeedFunction, ySpeedFunction, turnSpeedFunction;
+    private final Supplier<Boolean> fieldOrientedFunction; //Whether or not control is field oriented
+    private final SlewRateLimiter xLimiter, yLimiter, turnLimiter;
+
+    public SwerveTeleopCommand(SwerveSubsystem swerveSubsystem,
+            Supplier<Double> xSpeedFunction, Supplier<Double> ySpeedFunction, Supplier<Double> turnSpeedFunction,
+            Supplier<Boolean> fieldOrientedFunction) {
+        this.swerveSubsystem = swerveSubsystem;
+        this.xSpeedFunction = xSpeedFunction;
+        this.ySpeedFunction = ySpeedFunction;
+        this.turnSpeedFunction = turnSpeedFunction;
+        this.fieldOrientedFunction = fieldOrientedFunction;
+        this.xLimiter = new SlewRateLimiter(DriveConstants.kMaxAcceleration);
+        this.yLimiter = new SlewRateLimiter(DriveConstants.kMaxAcceleration);
+        this.turnLimiter = new SlewRateLimiter(DriveConstants.kMaxAngluarAcceleration);
+        addRequirements(swerveSubsystem);
+    }
+
+    @Override
+    public void initialize() {}
+
+    @Override
+    public void execute() {
+        //Get joystick inputs
```

Comment:
Trying to think this through.

We use a raw joystick input as a supplier in the constructor.

Then in the execute method, we are applying a deadband, 
applying a slew limiter,
and then sending that to the swerve kinematics.

We may have talked about this already, but I would like to see these functionalities separated.  
Do you have a plan for that?

Reply:
I would also like to separate this functionalities, to be writing as little actual code in the command as possible. However, this was the code included in the reference, so I wanted to hold the rework for the next PR.